server/schemas/schemaMatcher.js

```diff
@@ -0,0 +1,72 @@
+var csv = require("csv");
+var Readable = require("stream").Readable;
+var schemaModel = require("./schemaModel");
+var helpers = require("../config/helpers");
+
+//turns an array into a query which can be sent to mongoDB
+//via mongoose in order to find a schema template with the same fields.
+var queryBuilder = function(array) {
+  var obj = {};
+  for (var i = 0; i < array.length; i++) {
+    if (!obj.hasOwnProperty(array[i].toString()) && array[i].length) {
+      obj[array[i].toString()] = true;
+    }
+  }
+  var result = {};
+  for (var prop in obj) {
+    if (obj.hasOwnProperty(prop)) {
+      result["data." + prop] = {
+        $exists: true
+      };
+    }
+  }
+  return result;
+};
+
+var handle = function(buffer, callback) {
+  var stringifier = new csv.stringify();
+  var parser = new csv.parse();
+  var rs = new Readable({objectMode: true});
+  var validEntry = true;
+  var storage = {};
+  var columns;
+
+//We must define an internal _read function for our readable stream
+// that pulls data from the incoming file.
```

Comment:
this is a nitpick but you capitalize/space your comments inconsistently in many places